### PR TITLE
@task PDA-1489 Supporting OAuth 2.0 in swagger doc

### PR DESCRIPTION
**Pull request by lli-parasoft**

### CODE REVIEW

**Comment by lli-parasoft** [comment]
Once OAuth login the api will only send http request by OAuth, no matter Basic login or not.

**Comment by rwang-parasoft** [comment]
I have created a task[PDA-1490](https://jira.parasoft.com/browse/PDA-1490) to sign out the session in keycloak when logout in swagger ui.

**Comment by awang-parasoft** [comment]
Verified it works in Docker deployment.

File: src/main/java/com/parasoft/demoapp/config/OpenApiConfig.java [hunk 1]
```diff
@@ -21,18 +21,30 @@ public class OpenApiConfig {
...
                                                 .scheme("basic")
                                 )
+                                .addSecuritySchemes(oAuthSchemeName,
+                                        new SecurityScheme()
+                                                .name(oAuthSchemeName)
+                                                .type(SecurityScheme.Type.OAUTH2)
+                                                .flows(new OAuthFlows().authorizationCode(
+                                                        new OAuthFlow().tokenUrl(tokenUri)
+                                                                .authorizationUrl(authorizationUri)
+                                                                .scopes(new Scopes().addString("openid", "openid scope"))
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/demoapp/config/OpenApiConfig.java:45]: I think "openid" is enough. Why do we need "openid scope"?

**Comment by yshi-parasoft** [comment]
image

I always get this error. Do I need to do extra work to see it work?

File: src/main/java/com/parasoft/demoapp/config/OpenApiConfig.java (hunk 1 above)
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/demoapp/config/OpenApiConfig.java:45]: Maybe change "openid scope" to "for openid scope".

**Pull request merged**